app/javascript/packs/referral_charts.jsx

```diff
@@ -1,209 +1,93 @@
-import * as React from "react"
-import * as ReactDOM from "react-dom"
-import "babel-polyfill"
-import styled from "brave-ui/theme"
-import Select from "brave-ui/components/formControls/select"
-import ControlWrapper from "brave-ui/components/formControls/controlWrapper"
-import { PrimaryButton } from "../publishers/ReferralChartsStyle"
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import "babel-polyfill";
+import styled from "brave-ui/theme";
+import Select from "brave-ui/components/formControls/select";
+import ControlWrapper from "brave-ui/components/formControls/controlWrapper";
+import { PrimaryButton } from "../publishers/ReferralChartsStyle";
 // import "../publishers/dashboard_chart";
-import routes from "../views/routes"
-import Chart from "chart.js"
-import moment from "moment"
+import routes from "../views/routes";
+import Chart from "./chart/Chart";
+import { ThemeProvider } from "brave-ui/theme";
+import Theme from "brave-ui/theme/brave-default";
 
 export default class ReferralCharts extends React.Component {
-  constructor ( props ) {
-    super( props )
+  constructor(props) {
+    super(props);
     this.state = {
       referralCodes: this.props.referralCodes
-    }
-    this.selectMenuRef = React.createRef()
-    this.bindFunctions()
+    };
+    this.selectMenuRef = React.createRef();
+    this.bindFunctions();
   }
 
-  bindFunctions () {
-    this.viewReferralCodeStats = this.viewReferralCodeStats.bind( this )
+  bindFunctions() {
+    this.viewReferralCodeStats = this.viewReferralCodeStats.bind(this);
   }
 
-  createLabels ( startingDate ) {
```

Comment:
All these methods were moved to a subcomponent. It was a little bit of a cleaner React-esque way to handle these updates 🎉